### DISCUSSION TITLE
Long polling in using react-query

### DISCUSSION THREAD

**Discussion by rongdilin** [TanStack/query#3540] [Q&A]
How do we handle long polling xhr call from react-query? Do we have some example?

I saw a lot of examples to use refetchInterval(), but this will make an extra api call, and it's not long polling. The long polling call will be only one xhr call, and we get diff events like [progress](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/progress_event), load, loadEnd, etc.

**Comment by TkDodo** [comment]
You would probably want to set this up outside of react query and then react to events by populating the query client. This article could be helpful: https://tkdodo.eu/blog/using-web-sockets-with-react-query
